src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h

```diff
@@ -222,6 +222,7 @@ namespace sirf {
 		// abstract methods
 
 		virtual void empty() = 0;
+		virtual void take_over(MRAcquisitionData&) = 0;
```

Comment:
not related to PR?

Reply:
directly related - actually, the main difference from the previous approach is that now new data is created before the old one is dismissed, after which the object takes the ownership of the new data and discards the old one